File: ABC/201-250/229/b.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
  string a,b;
  cin >> a >> b;
  reverse(a.begin(),a.end());
  reverse(b.begin(),b.end());
  int n = min(a.size(),b.size());
  for(int i=0;i<n;i++){
    int ai = a[i]-'0';
    int bi = b[i]-'0';
    if(ai+bi>=10){
      cout << "Hard" << endl;
      return 0;
    }
  }
  cout << "Easy" << endl; 
  return 0;
}

File: ABC/201-250/229/b.py
def main():
  a,b = map(int,input().split())
  while a > 0 and b > 0:
    na = a%10
    nb = b%10
    if na + nb >= 10:
      print("Hard")
      return
    a //= 10
    b //= 10

  print("Easy")
if __name__ == '__main__':
  main()


